### PR TITLE
Fixup JavaCompiler a bit

### PR DESCRIPTION
**Pull request by tristan957**
* Add missing functions
* Use f-strings
* Fix mismatched parameter names

### CODE REVIEW

**Comment by tristan957** [comment]
cc @eli-schwartz. Wondering about good ways to test these changes. I couldn't find any documentation on the depfile suffix.

**Comment by tristan957** [comment]
Any suggestions on the best way to split the warn args between levels or just leave as is?

**Comment by tristan957** [comment]
Judging from your comment in the original issue, the depfile suffix is irrelevant. Maybe I should remove

File: mesonbuild/compilers/java.py [hunk 1]
```diff
@@ -39,6 +39,9 @@ def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoic
         super().__init__(exelist, version, for_machine, info, full_version=full_version)
         self.javarunner = 'java'
 
+    def get_warn_args(self, level: str) -> T.List[str]:
+        return ['-Xlint:all', '-Xdoclint:all']
```
Comment (dcbaker) [inline mesonbuild/compilers/java.py:43]: this seems to be ignoring the level parameter entirely?

Comment (tristan957) [reply mesonbuild/compilers/java.py:43]: yes. What do you think is the best course of action to use it. `1` is '-Xlint:all' and `2/3/...` include '-Xdoclint:all'?

Comment (jpakkane) [reply mesonbuild/compilers/java.py:43]: Seems reasonable. We can change it later if we get complaints.

**Pull request merged**